charts/k8ssandra/templates/_helpers.tpl

```diff
@@ -369,3 +369,82 @@ Cassandra image (Management API image and version tag).
 {{- (printf (get .Values.cassandra.versionImageMap .Values.cassandra.version)) }}
 {{- end }}
 {{- end }}
+
+{{/**
+SecurityContext helpers for pod and container scope.
+*/}}
+
+{{/**
+k8ssandra container securityContext defaults
+**/}}
+{{- define "k8ssandra.container.security-ctx-defaults" -}}
+{{ "securityContext:" -}}
+{{ "readOnlyRootFilesystem: true" | nindent 10 }}
+{{- end }}
+
+{{/**
+k8ssandra Cassandra pod's podSecurityContext defaults
+**/}}
+{{- define "k8ssandra.pod.pod-security-ctx-defaults" -}}
+{{ "securityContext: {}" -}}
+{{- end }}
+
+{{/**
+Container securityContext override checks.
+If specified securityContext, uses what is provided in values yaml.
+Otherwise, applies k8ssandra defaults as defined in security-ctx-defaults.
+**/}}
+{{- define "k8ssandra.container.security-ctx-base-config-init" -}}
```

Comment:
You can reduce all of these to a single template. You calling them with root scope. You just need to change the scope to the container in question. For example instead of,

```
{{ include "k8ssandra.container.security-ctx-base-config-init" . }}
```

you would instead have

```
{{ include "k8ssandra.container.security-ctx-base-config-init" .Values.baseConfig }}
```
This sets the variable scope of the template to an object that has a `securityContext` property.

Reply:
@jsanda
Are you thinking that for example, in the podTemplateSpec of cassdc, we would instead use this type of include?

```yaml
securityContext: {{ include "k8ssandra.container.security-ctx-base-config-init" .Values.cassandra.baseConfig.securityContext }}
```

Reply:
Yup. I had a mistake in my previous comment. Instead of `.Values.baseConfig` it should have been `.Values.cassandra.baseConfig`.

I would renamed the template to something more generic like `k8ssandra.container.security-ctx` since it would be used across multiple containers.

Reply:
@jsanda 

I did some digging into sprig and the helm capabilities.  I was, as you suggested, able to roll some of this up into less verbose logic for the helpers.  Thanks for suggesting this as I took a direct approach due to many errors that were cropping up.  I've added that update in the latest, and it looks easier to reason about IMO.

Right now, the security context for root read-only is relaxed for the `cassandra` and `mgmt-api` images to allow a root user to perform writes and reads.  Just those two containers' securityContext settings are impacted at this time.

For example, the `c* container` was failing to start due to errors such as:

```text
Starting Management API
cp: cannot create regular file '/opt/cassandra/conf/cassandra-env.sh': Read-only file system
cp: cannot create regular file '/opt/cassandra/conf/cassandra-rackdc.properties': Read-only file system
cp: cannot create regular file '/opt/cassandra/conf/cassandra.yaml': Read-only file system
cp: cannot create regular file '/opt/cassandra/conf/jmxremote.password': Read-only file system
cp: cannot create regular file '/opt/cassandra/conf/jvm.options': Read-only file system
cp: cannot create regular file '/opt/cassandra/conf/logback.xml': Read-only file system
```

As you know, we have some other tickets that require effort to get the images utiliized by k8ssandra to run as a non-root user.  For now, this is a starting point for getting k8ssandra securityContext configurations at the container and pod levels.  They are implemented along with the recent `volume` and `volume` mount changes as part of the overall security [epic](https://k8ssandra.atlassian.net/browse/K8SSAND-705).